test/test_collection.py

```diff
@@ -1707,7 +1707,8 @@ def test_find_one(self):
         self.assertTrue("hello" in db.test.find_one(projection=frozenset(["hello"])))
         self.assertTrue("hello" not in db.test.find_one(projection=frozenset(["foo"])))
 
-        self.assertEqual(["_id"], list(db.test.find_one(projection=[])))
+        self.assertEqual(["_id"], list(db.test.find_one(projection={'_id':
+            True})))
```

Comment:
Please add a new test for the explicitly empty projection case (ie projection={} and projection=[]).

Reply:
Done.

Reply:
Please add a test for `[]` as well.

Reply:
Done.